### PR TITLE
add parameter m

### LINKED ISSUE #7: Parameter -m

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #7

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -75,6 +77,10 @@ func main() {
 		regArr = append(regArr, re)
 	}
 
+	if *maxCountParameter != -1 {
+		counter = 0
+	}
+
```
Comment (Al2Klimov) [inline main.go]: What's this for?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -v -m 2 -e b <<EOF
a
a
a
EOF
```

... doesn't work as expected.

File: main.go (hunk 1 above)
Comment (justinmueller05082000) [reply main.go]: well i dont need to declare my counter var if my parameter is not set so i just need to define it and if my parameter is set i declare it

Comment (Al2Klimov) [reply main.go]: I mean: What's `counter = 0` for?

Comment (justinmueller05082000) [reply main.go]: counter is not declared yet so i dont know what would happen if do counter++ for var counter int

Comment (Al2Klimov) [reply main.go]: You don't know? Figure it out!

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -v -m 2 -e b <<EOF
a
b
a
EOF
```

... doesn't work as expected.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/parameter-m-7`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toGrep.txt:**

```
a
a
b
a
a
a
b
b
```

**Go code:**

**Terminal:** `go run . -m 2 -e 'a' < toGrep.txt`

**Output:**

```
a
a
```

**Grep command:**

**Terminal:**  `grep -m 2 -e 'a' < toGrep.txt`

**Output:**

```
a
a
```


**Conclusion:**
Both results are equal except for the colored highlighting in the terminal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**